cli/src/commands/diff.rs

```diff
@@ -0,0 +1,526 @@
+use clap::Parser as ClapParser;
+use codec::Decode;
+use color_eyre::eyre::eyre;
+use frame_metadata::RuntimeMetadataPrefixed;
+use std::collections::HashMap;
+use std::path::PathBuf;
+use std::str::FromStr;
+
+use crate::utils::FileOrUrl;
+use color_eyre::owo_colors::OwoColorize;
+use jsonrpsee::client_transport::ws::Uri;
+use scale_info::form::PortableForm;
+use scale_info::Variant;
+use subxt_codegen::utils::MetadataVersion;
+use subxt_metadata::{
+    ConstantMetadata, Metadata, PalletMetadata, RuntimeApiMetadata, StorageEntryMetadata,
+    StorageEntryType,
+};
+
+/// Explore the differences between two nodes
+///
+/// # Example
+/// ```
+/// subxt diff ./artifacts/polkadot_metadata_small.scale ./artifacts/polkadot_metadata_tiny.scale
+/// subxt diff ./artifacts/polkadot_metadata_small.scale wss://rpc.polkadot.io:443
+/// ```
+#[derive(Debug, ClapParser)]
+pub struct Opts {
+    node1: String,
```

Comment:
I would prefer to change this to `entry1: FileOrUrl` and `entry2: FileOrUrl` then clap will just use the `TryFrom` or `FromStr` impl and you don't have to deal with that yourself.